tap_chargebee/streams/base.py

```diff
@@ -175,23 +175,24 @@ def sync_data(self):
             
             to_write = self.get_stream_data(records)
             
-            if self.ENTITY == 'event':
-                for event in to_write:
-                    if event["event_type"] == 'plan_deleted':
-                        Util.plans.append(event['content']['plan'])
-                    elif event['event_type'] == 'addon_deleted':
-                        Util.addons.append(event['content']['addon'])
-                    elif event['event_type'] == 'coupon_deleted':
-                        Util.coupons.append(event['content']['coupon'])
-            if self.ENTITY == 'plan':
-                for plan in Util.plans:
-                    to_write.append(plan)
-            if self.ENTITY == 'addon':
-                for addon in Util.addons:
-                    to_write.append(addon)
-            if self.ENTITY == 'coupon':
-                for coupon in Util.coupons:
-                    to_write.append(coupon) 
+            if self.config.get('include_deleted') not in ['false','False', False]:
```

Comment:
I'm not sure why this behavior is behind the new flag. Shouldn't the flag only affect this logic by appending the events when they're "*_deleted" and the flag is true?

Reply:
Yes, it should like that. But for some of the streams `include_deleted` param isn't supported by API, so for those streams, tap collecting deleted records from the Event stream. 

If include_deleted=False then we won't need to collect these records from the event stream, hence I put this behavior behind the flag.